### DISCUSSION TITLE
Using Matcher for each sentence

### DISCUSSION THREAD

**Discussion by farrandi** [explosion/spaCy#9304] [Help: Coding & Implementations] [answered]
I am using token matching for my doc object and I notice it is matching tokens from different sentences too. 
Is there a way to use the matcher for only tokens within a sentence?

I am using this code right now, but I think it would significantly slow down the process
```
for sent in doc.sents:
    matches = matcher(sent, as_spans=True)
```

**Comment by polm** [comment, marked as answer]
There is not an option you can give the Matcher that will make it only match within sentences. I think the way you are doing this is the most straightforward way and shouldn't be too slow. Have you tried it to see how much slower it gets?

There is one other thing you could do. Matcher patterns can match on the `IS_SENT_START` attribute, so you could specify that for every token after the first that needs to be `False`. However I think specifying that would be more complicated and hard to understand.

**Reply by farrandi** [reply]
Ah I see. I tried measuring the time it took to process 100 documents and it was only 600 ms slower, so that is pretty decent.

Thanks!
